src/blocks/scratch3_procedures.js

```diff
@@ -13,46 +13,39 @@ class Scratch3ProcedureBlocks {
      */
     getPrimitives () {
         return {
-            procedures_definition: this.definition,
             procedures_call: this.call,
             argument_reporter_string_number: this.argumentReporterStringNumber,
             argument_reporter_boolean: this.argumentReporterBoolean
         };
     }
 
-    definition () {
-        // No-op: execute the blocks.
-    }
-
     call (args, util) {
-        if (!util.stackFrame.executed) {
```

Comment:
@cwillisf, @thisandagain, any idea why we would have had this? It seems it's back from when custom procedure calls were first introduced to the VM so I'm hesitant about removing it without fully understanding why it was there... (see @mzgoddard's corresponding comment in the PR description about executing the call function twice).

Reply:
My guess is dirty stack frame behaviour, with a pop out of a procedure after a loop and with isLoop having been unset. And so sequencer have treated exiting the procedure like it was a loop, left the stack at the procedure call id, and yielded.

Reply:
Looks like that `if` dates back to the initial version of the code from @tmickel back in October of 2016. I can't say for sure why it was written this way but I suspect @mzgoddard is right that it was meant to avoid running the same procedure again after it exits without advancing the thread.